### PR TITLE
[WIP] First version of How To

### LINKED ISSUE #128: Documentation: How to
I think we should have a form of "How to" documentation which answers questions like:
- How can I define the frames?
- How can I make an object appearing from the background?
- How can I move one object from A to B?
- How can I move one object based on another object?

### PR DESCRIPTION
**Pull request by Wikunia**
## PR Checklist

They are basically not relevant for documentation PRs I think :wink: 

**Link to relevant issue(s)**
Closes #128 

**How did you address these issues with this PR? What methods did you use?**
I've started with a `howto.md` where I've answered all the questions which are mentioned in the issue. We probably want to add more. Hope the general format makes sense.

### CODE REVIEW

File: docs/src/howto.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+function ground(args...)
+    background("black")
+    sethue("white")
+end
+
+video = Video(500, 500)
+
+javis(video, [
+    BackgroundAction(1:100, ground),
+    SNIPPETS_GO_HERE
```
Comment (TheCedarPrince) [inline docs/src/howto.md:19]: Maybe at this line say:

```
    SNIPPETS_GO_HERE # Replace this line with the provided snippet!
```

File: docs/src/howto.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+]; pathname="how_to.gif")
+```
+
+Each of the code snippets should be part of `SNIPPETS_GO_HERE`.
```
Comment (TheCedarPrince) [inline docs/src/howto.md:23]: Maybe change to:

```
should be part of `SNIPPETS_GO_HERE`.
```

--> should replace the variable `SNIPPETS_GO_HERE`

File: docs/src/howto.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+**Examples:**
+```julia
+Action(1:100, (args...)->circle(O, 50, :fill)),
+Action(:same, (args...)->circle(Point(100, 100), 20, :stroke)),
+Action((args...)->circle(Point(-100, 100), 20, :stroke))
+```
+
+so either use the symbol `:same` or just don't mention frames.
+
+The last option is to define frames relative to the previous frame. More preciously the end of the last frame.
```
Comment (TheCedarPrince) [inline docs/src/howto.md:63]: preciously --> precisely

File: docs/src/howto.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+## How can I define frames?
+
+There are currently three different ways to define frames inside Javis.
+The simplest one is to define the `UnitRange` like `1:100` as above such that the action is called for every frame from `1` to `100`.
+
+**Examples:**
+```julia
+Action(1:100, (args...)->circle(O, 50, :fill)),
+Action(1:50, (args...)->circle(O, 20, :stroke))
+```
```
Comment (TheCedarPrince) [inline docs/src/howto.md:50]: Maybe explain how this first action creates a ball at the origin of size 50 for the first hundred frames and the second action creates a ball the size of 20 for the first 20 frames and further explain that the balls would be on top of each other.

File: docs/src/howto.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+Action(Rel(1:50), (args...)->circle(Point(100, 100), 20, :stroke)),
+```
+
+This is the same as:
+```julia
+Action(1:50, (args...)->circle(O, 50, :fill)),
+Action(51:100, (args...)->circle(Point(100, 100), 20, :stroke)),
+```
+
+## How can I make an object appearing from the background?
```
Comment (TheCedarPrince) [inline docs/src/howto.md:77]: Maybe: How Can I Make an Object Fade in from the Background?

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Looks good so far. Aside from my general comments, do you think it might be smart to include the gifs of what the output should look like so people could quickly see if that topic is what they are looking for?

File: docs/src/howto.md (hunk 4 above)
Comment (Wikunia) [reply docs/src/howto.md:50]: I increased the radius of the second one. Thanks for pointing it out. I think in the "how to" we should only answer the question asked. At that point I assume they read at least the first tutorial and know the basics.

**Comment by Wikunia** [comment]
I haven't included the output gif for each of them as I think it is distracting as all examples are very straightforward. It should be easy to see what should happen and it should be fairly simple to test it out.

**Comment by TheCedarPrince** [comment]
I think that is a reasonable assumption to make @Wikunia . Feel free to squash and merge whenever. :D

**Pull request merged**